src/components/app/details/triggerView/CIMaterialModal.tsx

```diff
@@ -40,31 +51,44 @@ export const CIMaterialModal = ({
         [],
     )
 
-    return (
+    useEffect(() => {
+        if (props.isJobView && props.environmentLists?.length > 0) {
```

Comment:
Please add a condition of `selectedCIPipeline?.environmentId` and remove the default value  0 and `?` check from next line